notebooks/python/raw/ex_1.ipynb

```diff
@@ -131,7 +131,8 @@
    },
    "outputs": [],
    "source": [
-    "#q0.hint()"
+    "#_COMMENT_IF(PROD)_\n",
+    "q0.hint()"
```

Comment:
> If hint() is called but doesn't exist for a specific Problem, does it throw an exception?

No. Though it would be nice if we treated it as a CI failure if we have a call to `.hint()` in the notebook for a question that has no hint. Implementing that would be tricky though.

I basically see two forms of validation that these notebooks enable. One is the automated testing through CI, where we fail if we hit any exceptions. The other is just having a human open the notebook, press 'run all', and visually inspect the output to make sure it makes sense. We'll never be able to write automated tests to catch issues like the solution code being badly formatted or a failure message from check() being too confusing or verbose.

Having the calls to hint() and solution() uncommented in the raw notebook is mostly for the benefit of that second kind of validation.

Reply:
I agree that adding the `_COMMENT_IF_` macros every time is cumbersome. There is an argument for making the lesson preprocessor smart enough to identify patterns like `q1.hint()` and automatically comment them out in the prod notebooks, no macro required. But there's always a risk in adding more magic.

Reply:
Cool.  I've sometimes run .hint() once but then commented it manually.  I see the case against this, but I'm inclined to keep doing it personally.

As long we aren't doing anything structurally that assumes future authors will need a lot of this "#_COMMENT_IF(PROD)" stuff, it doesn't seem like a big deal... Adding magic to do that automatically doesn't seem worth the complexity.